pytensor/tensor/slinalg.py

```diff
@@ -1669,6 +1669,99 @@ def block_diag(*matrices: TensorVariable):
     return _block_diagonal_matrix(*matrices)
 
 
+def _to_banded_form(A, kl, ku):
+    """
+    Convert a full matrix A to LAPACK banded form for gbmv.
+
+    Parameters
+    ----------
+    A: np.ndarray
+        (m, n) banded matrix with nonzero values on the diagonals
+    kl: int
+        Number of nonzero lower diagonals of A
+    ku: int
+        Number of nonzero upper diagonals of A
+
+    Returns
+    -------
+    ab: np.ndarray
+        (kl + ku + 1, n) banded matrix suitable for LAPACK
+    """
+    A = np.asarray(A)
+    m, n = A.shape
+    ab = np.zeros((kl + ku + 1, n), dtype=A.dtype, order="C")
+
+    for i, k in enumerate(range(ku, -kl - 1, -1)):
+        padding = (k, 0) if k >= 0 else (0, -k)
+        diag = np.pad(np.diag(A, k=k), padding)
+        ab[i, :] = diag
+
+    return ab
```

Comment:
I imagine this explains most of the python overhead for small cases?

Reply:
one way or another we have to do that though as part of the cost of the Op. Unless we demand users have inputs ready in that form.

Reply:
Yeah it's fine, I was just thinking out loud.

Reply:
This rearrangement could be done symbolically in a wrapper Op that calls the blas Op (which expects things to be ready in the correct form)

It might also be better to do smart column indexing on `ab` instead of using `pad`

Reply:
Yeah it's similar to the Solve, in that you can also do it once and reuse many times possibly, but I think that's too much micro-optimization for now. We also don't want to autodiff through it